lib/engine/game/g_18_zoo/step/dividend.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module Engine
+  module Game
+    module G18ZOO
+      module Step
+        class Dividend < Engine::Step::Dividend
+          def dividend_options(entity)
+            revenue = @game.routes_revenue(routes)
+
+            dividend_types.map do |type|
+              [type, send(type, entity, revenue)]
+            end.to_h
+          end
+
+          def log_run_payout(entity, kind, revenue, action, payout)
+            unless Dividend::DIVIDEND_TYPES.include?(kind)
+              @log << "#{entity.name} runs for #{revenue} and pays #{action.kind}"
+            end
+
+            if payout[:corporation].positive?
+              @log << "#{entity.name} withholds #{@game.format_currency(payout[:corporation])}"
+            elsif payout[:per_share].zero?
+              @log << "#{entity.name} does not run"
+            end
+          end
+
+          def share_price_change(entity, revenue)
+            :right if revenue >= threshold(entity)
+          end
+
+          def withhold(_entity, revenue)
+            {
+              corporation: (revenue / 25).ceil,
+              per_share: 0,
+              share_direction: :left,
+              share_times: 1,
+              divs_to_corporation: 0,
+            }
+          end
+
+          def payout(entity, revenue)
+            {
+              corporation: 0,
+              per_share: payout_per_share(entity, revenue),
+              share_direction: share_price_change(entity, revenue),
```

Comment:
I find it strange to read `share_price_change`, since in this game's case it is only used for paying out, not withholding. Since `payout` is the only caller, and `share_price_change` is a one-liner, would deleting `share_price_change` and using this here work instead?

```suggestion
              share_direction: revenue >= threshold(entity) ? :right : nil,
```

Reply:
done in 6e5282e3